blog/2023-05-31-importing-data-using-axios.md

```diff
@@ -0,0 +1,218 @@
+  headers: {
+    "X-API-Key": API_KEY
+  }
+});
+
+console.log(response.data._embedded.languages)
+
+// The logged output looks like this:
+// [
+//   {
+//     id: 1002049001, <-- This is the language id
+//     name: "English",
+//     tag: "en",
+//     originalName: "English",
+//     flagEmoji: "🇬🇧",
+//     base: true
+//   },
+//   {
+//     id: 1005650001,  <-- This is the language id
+//     name: 'Spanish',
+//     tag: 'es',
+//     originalName: 'español',
+//     flagEmoji: '🇪🇸',
+//     base: false
+//     },
+// ];
+```
+
+Now, when we know the id of the desired language, we can assign it to the import language. For example, when we would like to
+assign the "import language" with id `1015429001` to the existing language with id `1005650001` (Spanish). To do so, hit
+["Pair existing language"](/api#tag/Import/operation/selectExistingLanguage_1) endpoint.
+
+```javascript
+  const response = await axios.put("https://app.tolgee.io/v2/projects/import/result/languages/1015429001/select-existing/1005650001", null, {
+  headers: {
+    "X-API-Key": API_KEY,
+  }
+})
+```
+
+When the server responds with 200, the language is assigned.
+
+You can also check this by accessing the [import result endpoint](/api#tag/Import/operation/getImportResult_1). `
+GET https://app.tolgee.io/v2/projects/import/result`
+
+## Step 3 - Resolving conflicts
+
+Sometimes when you add files to the Tolgee platform, there might be some conflicts.
+
+To find out which imported translations conflict with existing ones,
+use [this endpoint](https://tolgee.io/api#tag/Import/operation/getImportTranslations_1).
+`GET https://app.tolgee.io/v2/projects/import/result/languages/{languageId}/translations`
+
+The `languageId` parameter refers to "import language," not the "existing language"!
+
+To resolve these conflicts, you can use these endpoints:
+
+- Resolve conflict (override)
+- Resolve conflict (keep existing)
+- Resolve all translation conflicts (override)
+- Resolve all translation conflicts (keep existing)
+
+It's all documented [here](https://tolgee.io/api#tag/Import/operation/resolveTranslationSetOverride_1). So I won't cover
+this in detail.
+
+## Step 4 - Applying the import
+
+To finally apply the import, use the [`Apply` endpoint](/api#tag/Import/operation/applyImport_1).
+
+```javascript
+await axios.put("https://app.tolgee.io/v2/projects/import/apply", null, {
+  headers: {
+    "X-API-Key": API_KEY,
+  }
+})
+```
+
+When there are still some unresolved conflicts, you can force Tolgee to keep existing data or to override them using `forceMode` query params with possible values `OVERRIDE` or `KEEP`.
+
+When the response is OK, your data is imported. Yeeey!
+
+## (Bonus) Working with namespaces
+
+When you work with namespaces in your project, you might also want to select a namespace for the imported languages as
+well.
+This is possible using the [`Select namespace`](https://tolgee.io/api#tag/Import/operation/selectNamespace_1) endpoint.
```

Comment:
Shouldn't this section be before the Applying step? If the user works with namespaces, it would be great for them to know this before they try the final Applying step.